app/collins/controllers/actions/ipaddress/DeleteAction.scala

```diff
@@ -24,24 +25,31 @@ case class DeleteAction(
   handler: SecureController
 ) extends SecureAction(spec, handler) with AssetAction with ParamValidation {
 
-  case class ActionDataHolder(asset: Asset, pool: Option[String]) extends RequestDataHolder
+  case class ActionDataHolder(asset: Asset, pool: Option[String], address: Option[String]) extends RequestDataHolder
 
   val dataForm = Form(
-    "pool" -> validatedOptionalText(1)
+    Forms.tuple(
+      "pool" -> validatedOptionalText(1),
+      "address" -> validatedOptionalText(1)
+    )
   )
 
   override def validate(): Validation = withValidAsset(assetTag) { asset =>
-    val pool: Option[String] = dataForm.bindFromRequest()(request).fold(
-      err => None,
+    val (pool, address) : (Option[String], Option[String]) = dataForm.bindFromRequest()(request).fold(
+      err => (None, None),
       str => str
     )
-    Right(ActionDataHolder(asset, pool))
+    Right(ActionDataHolder(asset, pool, address))
   }
 
   override def execute(rd: RequestDataHolder) = Future {
     rd match {
-      case ActionDataHolder(asset, pool) =>
-        val deleted = IpAddresses.deleteByAssetAndPool(asset, pool)
+      case ActionDataHolder(asset, pool, address) =>
```

Comment:
Would it be more "scala-y" for a construct like the following?
```
val deleted = rd match {
  case ActionDataHolder(asset, None, Some(address)) => IpAddresses.deleteByAssetAndAddress(asset, address)
  case ActionDataHolder(asset, Some(pool), None) => IpAddresses.deleteByAssetAndPool(asset, pool)
  case _ => // throw an error cause you can only specify a pool, or an address, not neither or both
}
tattler.notice("Deleted %d IP addresses".format(deleted), asset)
ResponseData(Status.Ok, JsObject(Seq("DELETED" -> JsNumber(deleted))))
```

Reply:
Ah! I was trying to figure out how to do this. This looks much better I'll get this cleaned up.

Reply:
This turned into...
```
       case ActionDataHolder(asset, None, None) => // delete all addresses
      case ActionDataHolder(asset, None, Some(address)) => // delete single address
      case ActionDataHolder(asset, Some(pool), None) => // delete address in single pool
      case ActionDataHolder(asset, Some(pool), Some(address)) => // error thrown
```

The first case I handled because currently in the master branch running
```
curl --basic -X DELETE -u "blake:admin:first" -v 'http://localhost:9000/api/asset/tumblrtag300/addresses'
```
Will trigger a delete on all pools for an asset. If others think we should change that behavior I can switch it but just wanted to make sure that the API change was intentional.

Reply:
Great catch. I think we should document this behavior in the API docs (what happens when you specify none+none, pool, address, or both)